infrastructure/logging/src/main/java/tech/pegasys/teku/infrastructure/logging/Converter.java

```diff
@@ -14,22 +14,30 @@
 package tech.pegasys.teku.infrastructure.logging;
 
 import java.math.BigDecimal;
+import java.math.BigInteger;
 import java.math.RoundingMode;
 import org.apache.tuweni.units.bigints.UInt256;
 import org.web3j.utils.Convert;
+import org.web3j.utils.Convert.Unit;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
 
 public class Converter {
 
-  static BigDecimal gweiToEthFactor = BigDecimal.TEN.pow(18);
+  static BigDecimal gweiToEthFactor = BigDecimal.TEN.pow(9);
 
   public static String weiToEth(final UInt256 wei) {
     final BigDecimal result = Convert.fromWei(wei.toDecimalString(), Convert.Unit.ETHER);
     return result.setScale(6, RoundingMode.HALF_UP).toString();
   }
 
-  public static String gweiToEth(final UInt256 gwei) {
-    return new BigDecimal(gwei.toBigInteger())
+  public static String gweiToEth(final UInt64 gwei) {
+    return new BigDecimal(gwei.bigIntegerValue())
```

Comment:
nit: we could use `fromWei(toWei)` here or `weiToEth(toWei)` to exclude such errors in future

Reply:
I like that, implemented the second approach